# HG changeset patch
# User Jacob Dawid <jacob.dawid@googlemail.com>
# Date 1340274121 -7200
#      Thu Jun 21 12:22:01 2012 +0200
# Branch gui
# Node ID 84d1d9c035c70d028186bbdd190dda637ca0969c
# Parent  0d35ab1cf10cb91f81c969b2d37f2151417776a1
Changing the directory one step up now does not use the terminal.

* main-window.cc: Used event instead of sending text to terminal.

diff --git a/gui/src/main-window.cc b/gui/src/main-window.cc
--- a/gui/src/main-window.cc
+++ b/gui/src/main-window.cc
@@ -193,21 +193,21 @@ main_window::change_current_working_dire
   octave_link::instance ()
       ->post_event (new octave_change_directory_event (*this,
                     directory.toStdString ()));
 }
 
 void
 main_window::current_working_directory_up ()
 {
-  _terminal->sendText ("cd ..\n");
-  _terminal->setFocus ();
+  octave_link::instance ()
+      ->post_event (new octave_change_directory_event (*this, ".."));
+
 }
 
-
 void
 main_window::handle_entered_debug_mode ()
 {
   setWindowTitle ("Octave (Debugging)");
   _file_editor->handle_entered_debug_mode ();
   _debug_menu->setEnabled (true);
 }
 
